kernel/packages/shared/profiles/sagas.ts

```diff
@@ -697,22 +618,32 @@ export async function modifyAvatar(params: {
 
       newSnapshots[selector] = hash
       content.set(name, hash)
-      contentFile && files.push(contentFile)
+      contentFile && files.set(contentFile.name, contentFile.content)
     }
     newAvatar.snapshots = newSnapshots as Avatar['snapshots']
   }
   const newProfile = ensureServerFormat({ ...profile, avatar: newAvatar }, currentVersion)
 
-  const [data] = await buildDeployData(
-    identity,
+  return deploy(url, identity, { avatars: [newProfile] }, files)
+}
+
+async function deploy(url: string, identity: ExplorerIdentity, metadata: any, contentFiles: Map<string, Buffer>) {
+  // Build entity and group all files
+  const preparationData = await DeploymentBuilder.buildEntity(
+    EntityType.PROFILE,
     [identity.address],
-    {
-      avatars: [newProfile]
-    },
-    files,
-    content
+    contentFiles,
+    metadata
   )
-  return deploy(url, data)
+  // sign the entity id fetchMetaContentServer
+  const authChain = Authenticator.signPayload(identity, preparationData.entityId)
+  // Build the client
+  const baseUrl = url.replace('/content', '')
+  const catalyst = new CatalystClient(baseUrl, 'explorer-profile')
```

Comment:
You can use a `ContentClient` instead of a `CatalystClient`, and avoid removing the `/content`